internal/api/verify.go

```diff
@@ -696,12 +702,8 @@ func (a *API) verifyUserAndToken(conn *storage.Connection, params *VerifyParams,
 			sentAt = user.PhoneChangeSentAt
 			expectedToken = user.PhoneChangeToken
 		}
-		if config.Sms.IsTwilioVerifyProvider() {
-			if testOTP, ok := config.Sms.GetTestOTP(params.Phone, time.Now()); ok {
-				if params.Token == testOTP {
-					return user, nil
-				}
-			}
+
+		if !config.Hook.SendSMS.Enabled && config.Sms.IsTwilioVerifyProvider() {
```

Comment:
hmm why do we also need to check for whether the hook is enabled here? i guess it's because the hook always takes precedence over the provider? 

in the case of twilio verify, one can still use twilio verify through the hook and expect to use this verification path so i think we should remove it
```suggestion
		if config.Sms.IsTwilioVerifyProvider() {
```

Reply:
You can't use Twilio Verify in the hook at this stage. Twilio Verify is not a "send SMS" type of provider. If you have the hook enabled, you're using something that sends an actual code over to Twilio (which you can't do with Verify).

This whole thing is the fix that prevents people from using the hook if they have Twilio Verify setup as the phone provider.

Reply:
Twilio Verify would be used if we had a "generate OTP" and "verify OTP" hooks, which we don't.